react-router/history.d.ts

```diff
@@ -0,0 +1,183 @@
+// Type definitions for history v1.11.1
+// Project: https://github.com/rackt/history
+// Definitions by: Sergey Buturlakin <http://github.com/sergey-buturlakin>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+
+declare namespace HistoryModule {
+
+    // types based on https://github.com/rackt/history/blob/master/docs/Terms.md
+
+    type Action = string
+
+    type BeforeUnloadHook = () => string
+
+    type CreateHistory<T> = (options?: HistoryOptions) => T
+
+    type CreateHistoryEnhancer<T, E> = (createHistory: CreateHistory<T>) => CreateHistory<T & E>
+
+    interface History {
+        listenBefore(hook: TransitionHook): Function
+        listen(listener: LocationListener): Function
+        transitionTo(location: Location): void
+        pushState(state: LocationState, path: Path): void
+        replaceState(state: LocationState, path: Path): void
```

Comment:
Sorry, didn't see new methods you mentioned, thought you was talking about `pushState`.
Regarding new methods, in the base history interface `push` and `replace` have to be without query parameter. This parameter have to be introduced in HistoryQueries interface in overriden `push` and `replace` as it is with `pushState` and `replaceState`, but it seems from implementation that guys forget to override these new methods in `useQueries` module.